### PR TITLE
FAI-12648: Task board relationship hotspot

### PR DESCRIPTION
**Pull request by ted-faros**
## Description

Add boards issue tracking for Jira source.  State between runs is stored in new Poseidon endpoint `/accounts/<id>/state`.

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/jira-source/src/streams/faros_board_issues.ts [hunk 1]
```diff
@@ -58,6 +64,56 @@ export class FarosBoardIssues extends StreamWithBoardSlices {
...
+      yield issue;
+    }
+    await this.updateBoardIssueState(tracker);
+  }
+
+  private async updateBoardIssueState(
+    tracker: BoardIssueTracker
+  ): Promise<void> {
+    if (this.farosClient && this.config.source_id) {
+      this.logger.info('Updating board issue state in Faros');
```
Comment (ypc-faros) [inline sources/jira-source/src/streams/faros_board_issues.ts:77]: Let's do `debug` level

File: sources/jira-source/src/streams/faros_board_issues.ts [hunk 2]
```diff
@@ -58,6 +64,56 @@ export class FarosBoardIssues extends StreamWithBoardSlices {
...
+      }
+    }
+  }
+
+  private async createBoardIssueTracker(
+    boardId: string
+  ): Promise<BoardIssueTracker> {
+    let boardIssueState: any;
+    if (this.farosClient && this.config.source_id) {
+      this.logger.info('Loading board issue state from Faros');
```
Comment (ypc-faros) [inline sources/jira-source/src/streams/faros_board_issues.ts:100]: `debug` level

File: faros-airbyte-cdk/src/utils.ts [hunk 3]
```diff
@@ -177,6 +177,13 @@ const SOURCE_COMMON_PROPERTIES = {
     description: 'Enable debug mode',
     default: false,
   },
+  source_id: {
```
Comment (ypc-faros) [inline faros-airbyte-cdk/src/utils.ts:180]: Let's rename to `faros_source_id` for consistency with the destination

File: sources/jira-source/src/streams/faros_board_issues.ts (hunk 2 above)
Comment (ted-faros) [reply sources/jira-source/src/streams/faros_board_issues.ts]: fixed

**Review by chalenge — APPROVED** [review]
Nice

File: sources/jira-source/src/jira.ts [hunk 4]
```diff
@@ -68,6 +68,7 @@ export interface JiraConfig extends AirbyteConfig {
   readonly graph?: string;
   readonly requestedStreams?: Set<string>;
   readonly use_sprints_reverse_search?: boolean;
+  readonly use_faros_board_issue_tracker?: boolean;
```
Comment (ted-faros) [inline sources/jira-source/src/jira.ts:71]: @ypc-faros what else do I need to do to register this config and set it for the Jira feed instances?

Comment (ypc-faros) [reply sources/jira-source/src/jira.ts:71]: We need to add it to the spec: https://github.com/faros-ai/airbyte-connectors/blob/main/sources/jira-source/resources/spec.json

File: faros-airbyte-cdk/src/sources/streams/stream-base.ts [hunk 5]
```diff
@@ -189,6 +189,16 @@ export abstract class AirbyteStreamBase {
     return currentStreamState ?? {};
   }
 
+  /**
+   * Override to perform any action before reading slices.
+   */
+  async onBeforeReadSlices(): Promise<void> {}
```
Comment (ypc-faros) [inline faros-airbyte-cdk/src/sources/streams/stream-base.ts:195]: nit: `onBeforeRead`

File: faros-airbyte-cdk/src/sources/streams/stream-base.ts [hunk 6]
```diff
@@ -189,6 +189,16 @@ export abstract class AirbyteStreamBase {
...
+
+  /**
+   * Override to perform any action after reading all slices.
+   */
+  async onAfterReadSlices(): Promise<void> {}
```
Comment (ypc-faros) [inline faros-airbyte-cdk/src/sources/streams/stream-base.ts:200]: nit: `onAfterRead`

File: sources/jira-source/resources/spec.json [hunk 7]
```diff
@@ -246,6 +246,13 @@
         "type": "string",
         "title": "Organization ID",
         "description": "Atlassian organization ID for fetching teams. Required only when using Fetch Teams with Jira Cloud. See https://confluence.atlassian.com/jirakb/what-it-is-the-organization-id-and-where-to-find-it-1207189876.html for information on how to find it."
+      },
+      "use_faros_board_issue_tracker": {
+        "order": 32,
+        "type": "boolean",
+        "title": "Use Board Issue Tracker",
+        "description": "Use stateful board issue tracker to track issues on boards. This uses the Faros API to persist the state of boards between runs.",
```
Comment (chalenge) [inline sources/jira-source/resources/spec.json:254]: add needs `faros_source_id`?

File: faros-airbyte-cdk/src/utils.ts [hunk 8]
```diff
@@ -177,6 +177,12 @@ const SOURCE_COMMON_PROPERTIES = {
     description: 'Enable debug mode',
     default: false,
   },
+  faros_source_id: {
```
Comment (ted-faros) [inline faros-airbyte-cdk/src/utils.ts:180]: @chalenge I understand that adding here obviates need to add to specs of all sources.

File: sources/jira-source/resources/spec.json (hunk 7 above)
Comment (ted-faros) [reply sources/jira-source/resources/spec.json:254]: added to `SOURCE_COMMON_PROPERTIES`, so, no I think.

Comment (chalenge) [reply sources/jira-source/resources/spec.json:254]: I meant to the description of the field

File: faros-airbyte-cdk/src/utils.ts (hunk 8 above)
Comment (chalenge) [reply faros-airbyte-cdk/src/utils.ts:180]: see above

File: sources/jira-source/resources/spec.json (hunk 7 above)
Comment (ypc-faros) [reply sources/jira-source/resources/spec.json:254]: I think @chalenge means to note in the description that this requires the `faros_source_id` to be present...

**Review by chalenge — APPROVED** [review]

File: sources/jira-source/resources/spec.json (hunk 7 above)
Comment (ted-faros) [reply sources/jira-source/resources/spec.json:254]: ah!

**Review by ypc-faros — APPROVED** [review]

**Pull request merged**